internal/services/network/application_gateway_resource.go

```diff
@@ -2881,6 +3329,79 @@ func flattenApplicationGatewayHTTPListeners(input *[]network.ApplicationGatewayH
 	return results, nil
 }
 
+func flattenApplicationGatewayListeners(input *[]network.ApplicationGatewayListener) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for _, v := range *input {
+		output := map[string]interface{}{}
+
+		if v.ID != nil {
+			output["id"] = *v.ID
```

Comment:
Can we parse and then construct the ID, setting its `.ID()` value back to state here?